types/yup/index.d.ts

```diff
@@ -147,8 +147,8 @@ export interface StringSchema<T extends string | null | undefined = string | und
         arrayOfValues: ReadonlyArray<U | Ref>,
         message?: MixedLocale['oneOf'],
     ): StringSchema<MaintainOptionality<T, U>>;
+    test(name: string, message: TestOptionsMessage, test: TestFunction<T | undefined | null>): this;
```

Comment:
I would add a comment here explaining that `| undefined | null` is added intentionally because the test will still run for undefined/null input whether or not they're set to be allowed. Otherwise inevitably someone else will submit a PR to remove that.